File: w05/demo/swap.py
# swap.py
# Callum Howard, 2019

def swap(a, b):
    return (b, a)


a = 55
b = 77

print('a holds the number {}'.format(a))
print('b holds the number {}'.format(b))

print('swapping the numbers...')

# swap(a, b)
a, b = swap(a, b)

print('now a holds the number {}'.format(a))
print('now b holds the number {}'.format(b))

File: w05/demo/swap.cpp
// swap.cpp
// Callum Howard, 2017

#include <iostream>
#include <cassert>

void swap(int first, int second);

int main() {

    int a = 55;
    int b = 77;

    std::cout << "a holds the number " << a << '\n';
    std::cout << "b holds the number " << b << '\n';

    std::cout << "swapping the numbers...\n";

    swap(a, b);

    std::cout << "now a holds the number " << a << '\n';
    std::cout << "now b holds the number " << b << '\n';

    assert(a == 77);
    assert(b == 55);

    return EXIT_SUCCESS;
}

void swap(int first, int second) {
    int temp = first;
    first = second;
    second = temp;
}

